File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int uniqueIndex = 0, iter = 1;
        int len = nums.size();
        while(iter < len){
            if(nums[iter] == nums[uniqueIndex]){
                iter++;
            }else{
                nums[uniqueIndex+1] = nums[iter];
                uniqueIndex++;
                iter++;
            }
        }
        return uniqueIndex + 1;
    }
};

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def recur(self, nums: List[int], idx: int, ans = List[List[int]]) -> None:
        if idx == len(nums):
            ans.append(nums[:])
            return
        for i in range(idx, len(nums)):
            nums[i], nums[idx] = nums[idx], nums[i]
            self.recur(nums, idx + 1, ans)
            nums[i], nums[idx] = nums[idx], nums[i]

    def permute(self, nums: List[int]) -> List[List[int]]:
        ans = []
        self.recur(nums, 0, ans)
        return ans